src/validation/tests/this_keyword_validation_tests.rs

```diff
@@ -0,0 +1,468 @@
+use insta::assert_snapshot;
+use test_utils::parse_and_validate_buffered;
+
+#[test]
+fn this_pointer_arithmetic_is_ok() {
+    let diagnostics = parse_and_validate_buffered(
+        r#"
+        FUNCTION_BLOCK fb
+        VAR
+            a : INT;
+        END_VAR
+            // Pointer arithmetic with THIS
+            a := (THIS + 1)^ + 5;
```

Comment:
I hate this 🙃
Not sure if we want to validate against this, not too important for now though

Reply:
Yeah me too but it's technically correct so people will use it :see_no_evil: